src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1827,7 +1844,13 @@ public override void Dispose()
 
             public int[] ToArray()
             {
-                int[] array = new int[_end - _start];
+                int length = _end - _start;
+                if (length == 0)
```

Comment:
Is this reachable? I'm on a phone so I can't check fully, but I'm pretty sure zero size should cause Empty partition to be returned instead of an instance of this class